CiteProc.php

```diff
@@ -968,14 +968,20 @@ function render($data, $mode = NULL) {
     $text = '';
 
     if (($var = $this->variable) && isset($data->{$var})) {
-      $date = $data->{$var}->{'date-parts'}[0];
-      foreach ($this->elements as $element) {
-        $date_parts[] = $element->render($date, $mode);
+      if (isset($data->{$var}->{'date-parts'})) {
+        $date = $data->{$var}->{'date-parts'}[0];
+        foreach ($this->elements as $element) {
+          $date_parts[] = $element->render($date, $mode);
+        }
+        $text = implode('', $date_parts);
+        return $this->format($text);
+      }
+      else if (isset($data->{$var}->{'literal'})) {
+        return $this->format($text);
```

Comment:
@qadan is this part planned? I might be missing something again, but it appears if there is a literal you just format a blank string. This makes sense as there is no date to parse, so you display nothing. Just not sure why you bother to pass the empty string to the format function?